application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksSystem.java

```diff
@@ -0,0 +1,185 @@
+            .asMatchPredicate();
+    }
+
+    /**
+     * Accepts the consumers for used for requesting the list and remove paginations.
+     *
+     * @param listPaginationConsumer The list pagination handler
+     * @param removePaginationConsumer The remove pagination handler
+     */
+    void acceptPaginationConsumers(Consumer<GenericCommandInteractionEvent> listPaginationConsumer,
+            Consumer<GenericCommandInteractionEvent> removePaginationConsumer) {
+        this.listPaginationConsumer = listPaginationConsumer;
+        this.removePaginationConsumer = removePaginationConsumer;
+    }
+
+    /**
+     * Requests a list pagination from the {@link BookmarksPaginatorInteractor}.
+     *
+     * @param event The command interaction event
+     */
+    void requestListPagination(GenericCommandInteractionEvent event) {
+        Objects.requireNonNull(listPaginationConsumer, "No list pagination consumer was provided");
+
+        listPaginationConsumer.accept(event);
+    }
+
+    /**
+     * Requests a remove pagination from the {@link BookmarksPaginatorInteractor}.
+     *
+     * @param event The command interaction event
+     */
+    void requestRemovePagination(GenericCommandInteractionEvent event) {
+        Objects.requireNonNull(removePaginationConsumer,
+                "No remove pagination consumer was provided");
+
+        removePaginationConsumer.accept(event);
+    }
+
+    boolean isHelpThread(MessageChannelUnion channel) {
+        if (channel.getType() != ChannelType.GUILD_PUBLIC_THREAD) {
+            return false;
+        }
+
+        ThreadChannel threadChannel = channel.asThreadChannel();
+        String parentChannelName = threadChannel.getParentMessageChannel().getName();
+
+        return isOverviewChannelName.test(parentChannelName);
+    }
+
+    boolean didUserBookmarkChannel(long userID, long channelID) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(userID), BOOKMARKS.CHANNEL_ID.eq(channelID))
+            .fetch()
+            .isNotEmpty());
+    }
+
+    void addBookmark(long authorID, long channelID, @Nullable String note) {
+        database.write(context -> context.newRecord(BOOKMARKS)
+            .setAuthorId(authorID)
+            .setChannelId(channelID)
+            .setCreatedAt(Instant.now())
+            .setNote(note)
+            .setScheduledRemovalAt(null)
+            .insert());
+    }
+
+    List<BookmarksRecord> getUsersBookmarks(long authorID) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .orderBy(BOOKMARKS.CREATED_AT.desc())
+            .fetch());
+    }
+
+    void removeBookmarks(long authorID, List<Long> channelIDs) {
+        database.write(context -> context.deleteFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID), BOOKMARKS.CHANNEL_ID.in(channelIDs))
```

Comment:
use a `Set` for the `in` check, not a list.

Reply:
Changed. It is now using a Set. With this however i changed a bit more things like method return types, parameters and names so bookmarksToRemoveChannelIDs is now Set only